### ISSUE TITLE
feat: make `false` link in `splitLink` optional

### ISSUE THREAD

**Issue by rhenriquez28** [trpc/trpc#2929]
### Describe the feature you'd like to request

While I was doing the Link documentation overhaul, I noticed in the [Apollo link docs](https://www.apollographql.com/docs/react/api/link/introduction/#directional-composition) that their `split` implementation has the `false` link as optional. I thought it would be cool to add that functionality to tRPC's own `splitLink` so that in the case the user just wants a conditional link they can use it without supplying a fallback.

In simpler terms, right now `splitLink` just provides an "if else" flow, and I would like it if it could also offer the "if" flow, without the else.

### Describe the solution you'd like to see

Basic implementation would be something like:

```ts
export function splitLink<TRouter extends AnyRouter = AnyRouter>(opts: {
  condition: (op: Operation) => boolean;
  /**
   * The link to execute next if the test function returns `true`.
   */
  true: TRPCLink<TRouter> | TRPCLink<TRouter>[];
  /**
   * The link to execute next if the test function returns `false`.
   */
  false?: TRPCLink<TRouter> | TRPCLink<TRouter>[];
}): TRPCLink<TRouter> {
  return (runtime) => {
    const yes = asArray(opts.true).map((link) => link(runtime));
    const no = opts.false
      ? asArray(opts.false).map((link) => link(runtime))
      : [];
    return (props) => {
      return observable((observer) => {
        const passesCondition = opts.condition(props.op);
        if (passesCondition === false && !opts.false) {
          return props.next(props.op).subscribe(observer);
        }
        const links = passesCondition ? yes : no;
        return createChain({ op: props.op, links }).subscribe(observer);
      });
    };
  };
}
```

### Desribe alternate solutions

Not add it and leave it as is.

### Additional information

Tests would need to be added to validate that it works properly.
Documentation would need to be modified

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

**Comment by KATT** [comment]
Sorry, I missed your feature request the other day. 

I think this is a really solid contribution and you did a great job, but I don't think we should have this functionality. 

I think it's better to explicitly define a link that throws than it is to allow optionality. It's one of those features that might cause unexpected behavior because the user simply forgot the `false`-link

_Originally posted by @KATT in https://github.com/trpc/trpc/pull/2968#pullrequestreview-1137806563_

**Comment by KATT** [comment]
@sachinraja @jlalmes if any of you disagree with my statement, feel free to reopen it.

**Comment by rhenriquez28** [comment]
I understand. However, I think one thing that could be salvaged from my PR would be the added description of `splitLink` in the docs. I didn't realize that when you used `splitLink` you actually created a whole new link chain instead of working as a normal "if else" block.

I thought `splitLink` could be used like this: 
```ts"
const client = createTRPCProxyClient<AppRouter>({
  links: [
    splitLink({
      condition(op) {
        return op.context.condition === true;
      },
      true: linkOne(),
      false: linkTwo(),
    }),
linkThree()
  ],
});
```

However, after doing the tests for my modification, I realized this is not the case. There's no explanation of that anywhere in the docs

**Comment by KATT** [comment]
> I understand. However, I think one thing that could be salvaged from my PR would be the added description of `splitLink` in the docs. I didn't realize that when you used `splitLink` you actually created a whole new link chain instead of working as a normal "if else" block.
> 
> I thought `splitLink` could be used like this:
> 
> ```
> const client = createTRPCProxyClient<AppRouter>({
>   links: [
>     splitLink({
>       condition(op) {
>         return op.context.condition === true;
>       },
>       true: linkOne(),
>       false: linkTwo(),
>     }),
> linkThree()
>   ],
> });
> ```
> 
> However, after doing the tests for my modification, I realized this is not the case. There's no explanation of that anywhere in the docs

Feel free to cherry pick the docs improvements, I really love the work you've been doing there!
